Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockIDownloaderHelper.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Moq;
+
+namespace SonarScanner.MSBuild.PreProcessor.Test.Infrastructure
+{
+    internal static class MockIDownloaderHelper
+    {
+        /// <summary>
+        /// Create basic mocked instance of <see cref="IDownloader"/> with
+        /// </summary>
+        /// <returns>The mocked instance <see cref="IDownloader"/>.</returns>
+        public static Mock<IDownloader> CreateMock()
+        {
+            var mock = new Mock<IDownloader>(MockBehavior.Strict);
+            mock.Setup(x => x.Dispose());
+
+            return mock;
+        }
+
+        /// <summary>
+        /// Setup the <see cref="IDownloader.TryDownloadIfExists"/> method for the current mocked <see cref="IDownloader"/>.
+        /// </summary>
+        /// <param name="mock">The mocked <see cref="IDownloader"/>.</param>
+        /// <param name="requestUri"><see cref="Uri"/> of the request resource, if null, will match any url.</param>
+        /// <param name="response">Response content, if null, mocked response will be: <code>(false, null)</code></param>
+        /// <returns>The mocked <see cref="IDownloader"/>.</returns>
```

Comment:
I would remove all the comments from the class. The names are good enough to make the code understandable.